src/github.com/matrix-org/dendrite/clientapi/routing/createroom.go

```diff
@@ -43,10 +42,30 @@ type createRoomRequest struct {
 	Topic           string                 `json:"topic"`
 	Preset          string                 `json:"preset"`
 	CreationContent map[string]interface{} `json:"creation_content"`
-	InitialState    json.RawMessage        `json:"initial_state"` // TODO
+	InitialState    []fledglingEvent       `json:"initial_state"`
 	RoomAliasName   string                 `json:"room_alias_name"`
+	GuestCanJoin    bool                   `json:"guest_can_join"`
 }
 
+const (
+	presetPrivateChat        = "private_chat"
+	presetTrustedPrivateChat = "trusted_private_chat"
+	presetPublicChat         = "public_chat"
+)
+
+const (
+	joinRulePublic = "public"
+	// joinRuleCanJoin = "can_join"
+	joinRuleInvite = "invite"
+	// joinRuleKnock   = "knock"
+	// joinRulePrivate = "private"
+)
+const (
+	historyVisibilityShared = "shared"
+	// historyVisibilityWorldReadable = "world_readable"
+	// historyVisibilityInvited       = "invited"
```

Comment:
Would be good to put a `// TODO: These should be implemented once history visibility is implemented`